src/components/markdown-renderer/hooks/use-markdown-extensions.ts

```diff
@@ -5,98 +5,57 @@
+import { LinkifyFixMarkdownExtension } from '../extensions/linkify-fix/linkify-fix-markdown-extension'
+import { LinkAdjustmentMarkdownExtension } from '../extensions/link-replacer/link-adjustment-markdown-extension'
+import { EmojiMarkdownExtension } from '../extensions/emoji/emoji-markdown-extension'
+import { LinemarkerMarkdownExtension } from '../extensions/linemarker/linemarker-markdown-extension'
+import { TableOfContentsMarkdownExtension } from '../extensions/table-of-contents-markdown-extension'
+import { ImagePlaceholderMarkdownExtension } from '../extensions/image-placeholder/image-placeholder-markdown-extension'
+import type { MarkdownRendererExtension } from '../extensions/base/markdown-renderer-extension'
+import { useExtensionEventEmitter } from './use-extension-event-emitter'
+import { extraAppExtensions } from '../../../extensions/extra-integrations/extra-app-extensions'
+import { ProxyImageMarkdownExtension } from '../extensions/image/proxy-image-markdown-extension'
+
+const extraRendererExtensions = extraAppExtensions.flatMap((value) => value.buildMarkdownRendererExtensions())
 
 /**
- * Provides a list of {@link MarkdownExtension markdown extensions} that is a combination of the common extensions and the given additional.
+ * Provides a list of {@link MarkdownRendererExtension markdown extensions} that is a combination of the common extensions and the given additional.
  *
  * @param baseUrl The base url for the {@link LinkAdjustmentMarkdownExtension}
  * @param currentLineMarkers A {@link MutableRefObject reference} to {@link LineMarkers} for the {@link LinemarkerMarkdownExtension}
  * @param additionalExtensions The additional extensions that should be included in the list
- * @param onTaskCheckedChange The checkbox click callback for the {@link TaskListMarkdownExtension}
- * @param onImageClick The image click callback for the {@link ProxyImageMarkdownExtension}
- * @param onTocChange The toc-changed callback for the {@link TableOfContentsMarkdownExtension}
  * @return The created list of markdown extensions
  */
 export const useMarkdownExtensions = (
   baseUrl: string,
   currentLineMarkers: MutableRefObject<LineMarkers[] | undefined> | undefined,
-  additionalExtensions: MarkdownExtension[],
-  onTaskCheckedChange?: (lineInMarkdown: number, checked: boolean) => void,
-  onImageClick?: ImageClickHandler,
-  onTocChange?: (ast?: TocAst) => void
-): MarkdownExtension[] => {
-  const toc = useRef<TocAst | undefined>(undefined)
-  useOnRefChange(toc, onTocChange)
-
+  additionalExtensions: MarkdownRendererExtension[]
+): MarkdownRendererExtension[] => {
+  const extensionEventEmitter = useExtensionEventEmitter()
+  //replace with global list
   return useMemo(() => {
     return [
-      new TableOfContentsMarkdownExtension((ast?: TocAst) => (toc.current = ast)),
+      ...extraRendererExtensions,
       ...additionalExtensions,
```

Comment:
I find the difference between `extraRendererExtensions` and `additionalExtensions` a bit puzzeling…

Reply:
I renamed the extra extensions to optional.